tools/release/download_github_artifacts.py

```diff
@@ -0,0 +1,154 @@
+artifactUrl = args.artifactUrl
+if not artifactUrl.startswith('https://github.com/'):
+    print("❌ Invalid parameter --artifactUrl %s. Must start with 'https://github.com/'" % artifactUrl)
+    exit(1)
+if "/artifacts/" not in artifactUrl:
+    print("❌ Invalid parameter --artifactUrl %s. Must contain '/artifacts/'" % artifactUrl)
+    exit(1)
+artifactItems = artifactUrl.split("/")
+if len(artifactItems) != 9:
+    print("❌ Invalid parameter --artifactUrl %s. Please check the format." % (artifactUrl))
+    exit(1)
+
+gitHubRepoOwner = artifactItems[3]
+gitHubRepo = artifactItems[4]
+artifactId = artifactItems[8]
+
+if args.verbose:
+    print("gitHubRepoOwner: %s, gitHubRepo: %s, artifactId: %s" % (gitHubRepoOwner, gitHubRepo, artifactId))
+
+headers = {
+   'Authorization': "Bearer %s" % args.token,
+   'Accept': 'application/vnd.github+json'
+}
+base_url = "https://api.github.com/repos/%s/%s/actions/artifacts/%s" % (gitHubRepoOwner, gitHubRepo, artifactId)
+
+### Fetch build state
+
+print("Getting artifacts data of project '%s/%s' artifactId '%s'..." % (gitHubRepoOwner, gitHubRepo, artifactId))
+
+if args.verbose:
+    print("Url: %s" % base_url)
+
+r = requests.get(base_url, headers=headers)
+data = json.loads(r.content.decode())
+
+if args.verbose:
+    print("Json data:")
+    print(data)
+
+if args.verbose:
+    print("Create subfolder %s to download artifacts..." % artifactId)
+
+if args.directory == "":
+    targetDir = artifactId
+else:
+    targetDir = args.directory
+
+if not args.simulate:
+    os.makedirs(targetDir, exist_ok=True)
+
+url = data.get("archive_download_url")
+if args.filename is not None:
+    filename = args.filename
+else:
+    filename = data.get("name") + ".zip"
+
+## Print some info about the artifact origin
+commitLink = "https://github.com/%s/%s/commit/%s" % (gitHubRepoOwner, gitHubRepo, data.get("workflow_run").get("head_sha"))
+print("Preparing to download artifact `%s`, built from branch: `%s` (commit %s)" % (data.get("name"), data.get("workflow_run").get("head_branch"), commitLink))
+
+if args.verbose:
+    print()
+    print("Artifact url: %s" % url)
+
+target = targetDir + "/" + filename
+sizeInBytes = data.get("size_in_bytes")
+print("Downloading %s to '%s' (file size is %s bytes, this may take a while)..." % (filename, targetDir, sizeInBytes))
+if not args.simulate:
+    # open file to write in binary mode
+    with open(target, "wb") as file:
+        # get request
+        response = requests.get(url, headers=headers)
+        # write to file
+        file.write(response.content)
+    print("Verifying file size...")
+    # get the file size
+    size = os.path.getsize(target)
+    if sizeInBytes != size:
+        # error = True
+        print("Warning, file size mismatch: expecting %s and get %s. This is just a warning for now..." % (sizeInBytes, size))
+
+if error:
+    print("❌ Error(s) occurred, please check the log")
+    exit(1)
+else:
+    print("Done!")
```

Comment:
This file is a copy of https://github.com/vector-im/element-android/blob/develop/tools/release/download_github_artifacts.py